renderer/reducers/address.js

```diff
@@ -67,15 +67,23 @@ export const initAddresses = () => async (dispatch, getState) => {
     const addresses = get(node, 'addresses', {})
 
     // Ensure that we have an address for all supported address types.
-    await Promise.all(
-      Object.keys(ADDRESS_TYPES).map(async type => {
-        if (!addresses[type]) {
-          return dispatch(newAddress(type))
-        }
-        return null
-      })
-    )
-    dispatch({ type: FETCH_ADDRESSES_SUCCESS, addresses })
+    const fetchAddresses = async types => {
+      const data = {}
+      await Promise.all(
+        types.map(async type => {
+          if (addresses[type]) {
+            data[type] = addresses[type]
+            return
```

Comment:
maybe just add else? probably no sense in using `return` in map

Reply:
updated